app/model/asana.js

```diff
@@ -64,11 +68,83 @@ const Asana = {
+    },
+    printColInfo() {
+        for (let column in this.columnTasks) {
+            console.log("Col ID: " + column)
+            this.columnTasks[column].forEach(task => {
+                console.log("\tTask Name: " + task.name)
+            })
+        }
+    },
+    async updateTasks() {
+        if (!this.updatingTasks) {
+            this.t2 = performance.now()
+            this.updatingTasks = true;
+            let timePassedSinceLastUpdate = this.t2 - this.t1;
+            let taskFields = 'custom_fields,tags.name,tags.color,memberships.section.name,memberships.project.name,name,assignee.photo,assignee.name,assignee.email,due_on,modified_at,html_notes,notes,stories'
+            let url = `https://app.asana.com/api/1.0/tasks?modified_since=${new Date(Date.now() - timePassedSinceLastUpdate).toISOString()}&project=${this.projectId}&opt_fields=${taskFields}`
+            let response = await x.request({ url: url }).then((response) => {
+                return response
+            })
+            for (let task of response.data) {
+                for (let membership of task.memberships) {
+                    if (membership.project.gid === this.projectId) {
+                        let ss = this.getSectionAndSwimlane(membership.section)
+                        if (ss) {
+                            const outgoingColID = this.tasks[task.gid].memberships[0].section.gid;
+                            const incomingColID = membership.section.gid;
+
+                            if (outgoingColID === incomingColID) {
+                                this.tasks[task.gid] = task;
+                            } else {
+                                this.moveTask(task, outgoingColID, incomingColID, false);
+                            }                            
+                        }
+                    }
+                }
+            }
+            for (let key in this.tasks) {
+                let task = this.tasks[key]
+                this.rejiggerFields(task)
+            }
+            this.updatingTasks = false;
+            this.t1 = performance.now()
+        }
+    },
+    moveTask(task, outgoingColID, incomingColID, fromLocal) {
+
+        if (!this.columnTasks[task.memberships[0].section.gid]) {
+            this.columnTasks[task.memberships[0].section.gid] = []
+        }
+
+        let taskExistsInColumn = false;
+        this.columnTasks[incomingColID].forEach(colTask => {
+            if (colTask.gid === task.gid) {
+                taskExistsInColumn = true;
+            }
+        })
+        if (!taskExistsInColumn) {
+            this.columnTasks[incomingColID].push(task)
+            if (!fromLocal) {
+                this.tasks[task.gid] = task;
+                this.sectionMeta[this.getSectionAndSwimlane(this.columnTasks[incomingColID][0].memberships[0].section).sectionName].count++
+            }
+        }
+
+        let taskIndex = -1;
+        this.columnTasks[outgoingColID].forEach((colTask, i) => {
```

Comment:
You imported underscore but you aren't using it in a place like this (which is a very good place for it)